### PR TITLE
DAgent Kouros Integration with gRPC implementation

### PR DESCRIPTION
**Pull request by vishal-shakya-ssir**
This commit has below changes 
- DAgent Kouros Integration with gRPC implementation of all POS commands  (except Volume and QOS commands)

### CODE REVIEW

File: src/pnconnector/src/routers/m9k/api/ibofos/volume.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ import (
 	"pnconnector/src/errors"
 	"pnconnector/src/influxdb"
 	"pnconnector/src/routers/m9k/model"
+    //"kouros/model"
```
Comment (aswinssir) [inline src/pnconnector/src/routers/m9k/api/ibofos/volume.go:38]: Please remove the commented code

File: src/kouros/pos/pos_manager.go [hunk 2]
```diff
@@ -161,4 +161,17 @@ type POSManager interface {
 	// Stop the collection of telemetry data in PoseidonOS
 	// The function takes a protobuf format as parameter and returns response in protobuf format
 	StopTelemetry() (*pb.StopTelemetryResponse, error)
+
+
+    // Create the volume in PoseidonOS Array
+    // The function takes a protobuf format as parameter and returns response in protobuf format
+    CreateVolume(param *pb.CreateVolumeRequest_Param) (*pb.CreateVolumeResponse, error)
```
Comment (aswinssir) [inline src/kouros/pos/pos_manager.go:168]: Looks like the code is not indendented properly. Please run go fmt to format it

File: src/kouros/model/model.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+	Normal    string `json:"normal,omitempty"`
+	Urgent    string `json:"urgent,omitempty"`
+}
+
+type BuildInfo struct {
+	GitHash   string `json:"githash"`
+	BuildTime string `json:"buildTime"`
+}
+
+//type SMART struct {
```
Comment (aswinssir) [inline src/kouros/model/model.go:241]: Please remove commented code

File: src/kouros/ha/postgres/postgres_service.go [hunk 4]
```diff
@@ -3,7 +3,7 @@ package postgres
 import (
 	sql "database/sql"
 	"fmt"
-	pb "kouros/pos/grpc/api"
+    pb "kouros/api"
```
Comment (aswinssir) [inline src/kouros/ha/postgres/postgres_service.go:6]: Please correct the indendation

File: src/dagent/src/routers/m9k/api/responseBuilder.go [hunk 5]
```diff
@@ -33,28 +33,27 @@
...
-	"pnconnector/src/log"
-	iBoFOS "pnconnector/src/routers/m9k/api/ibofos"
-	"pnconnector/src/routers/m9k/model"
-	//"pnconnector/src/setting"
+	"kouros/log"
+	"kouros/model"
 	"github.com/gin-gonic/gin"
 	"net/http"
-	"pnconnector/src/util"
+    "dagent/src/routers/m9k/api/caller"
```
Comment (aswinssir) [inline src/dagent/src/routers/m9k/api/responseBuilder.go:40]: Please run go fmt to correct indendation

File: src/dagent/src/routers/m9k/api/dagent/multiVolHandler.go [hunk 6]
```diff
@@ -44,6 +44,7 @@ import (
 	"pnconnector/src/log"
 	iBoFOS "pnconnector/src/routers/m9k/api/ibofos"
 	"pnconnector/src/routers/m9k/model"
+    //"kouros/model"
```
Comment (aswinssir) [inline src/dagent/src/routers/m9k/api/dagent/multiVolHandler.go:47]: Please remove the commented code

File: src/dagent/src/routers/m9k/api/dagent/general.go [hunk 7]
```diff
@@ -67,7 +71,10 @@ func HeartBeat(xrId string, param interface{}) (model.Response, error) {
 func updateSuccessTime(xrId string) int64 {
 	if LastSuccessTime+MAXAGE < time.Now().UTC().Unix() {
 		param := model.DeviceParam{}
-		_, res, _ := iBoFOS.IBoFOSInfo(xrId, param)
+        posMngr,_ := kouros.NewPOSManager(pos.GRPC)
+        posMngr.Init(model.RequesterName,setting.Config.Server.IBoF.IP+":"+setting.Config.Server.IBoF.GrpcPort)
+		res, _ := caller.CallGetSystemInfo(xrId, param,posMngr)
+        fmt.Println("caller.CallGetSystemInfo ",res," time ",res.LastSuccessTime)
```
Comment (aswinssir) [inline src/dagent/src/routers/m9k/api/dagent/general.go:77]: Please remove this message if it was added for debugging purpose

**Review by aswinssir — CHANGES_REQUESTED** [review]
Please format the code using go fmt and remove the commented code

File: src/dagent/script/build_dagent.sh [hunk 8]
```diff
@@ -6,7 +6,10 @@ cd ..
 rm -rf ./vendor/pnconnector
 cp -rf ../pnconnector ./vendor/
```
Comment (manoj22-k) [inline src/dagent/script/build_dagent.sh:7]: Are these two lines required?

File: src/pnconnector/src/routers/m9k/api/ibofos/volume.go (hunk 1 above)
Comment (vishal-shakya-ssir) [reply src/pnconnector/src/routers/m9k/api/ibofos/volume.go]: Thanks for pointing out, I have removed the commented code

File: src/kouros/pos/pos_manager.go (hunk 2 above)
Comment (vishal-shakya-ssir) [reply src/kouros/pos/pos_manager.go]: Thanks for pointing out, formatting is done

File: src/dagent/script/build_dagent.sh (hunk 8 above)
Comment (vishal-shakya-ssir) [reply src/dagent/script/build_dagent.sh:7]: yes, we want to copy fresh pnconnector directory inside vendor directory

**Review by aswinssir — APPROVED** [review]
LGTM

**Review by manoj22-k — APPROVED** [review]
LGTM

**Pull request merged**